engines/director/lingo/xlibs/filexobj.cpp

```diff
@@ -0,0 +1,139 @@
+﻿/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "common/system.h"
+
+#include "director/director.h"
+#include "director/lingo/lingo.h"
+#include "director/lingo/lingo-object.h"
+#include "director/lingo/lingo-utils.h"
+#include "director/lingo/xlibs/filexobj.h"
+
+/**************************************************
+ *
+ * USED IN:
+ * Karma: Curse of the 12 Caves
+ *
+ **************************************************/
+
+/*
```

Comment:
This XObject is just the standard FileIO one we already have in fileio.cpp. I'd recommend removing this stub and instead adding a filename to the list in fileio.cpp.

Reply:
When I do that as you proposed

`const char *FileIO::fileNames[] = {
	"FileIO",
	"shFILEIO", // TD loads this up using openXLib("@:shFILEIO.DLL")
	"FILE",
	nullptr
};`

I'm getting this warning:
WARNING: BUILDBOT: Uncaught Lingo error: Call to undefined handler 'FILE'. Dropping 3 stack items!

Whereas this does not appear with my stub.

Is it possibly because "Karma: Curse of the 12 Caves" is a D4 game, and the DLL is a D4 DLL?
I saw that the methods are versioned with "200" but the game uses "404".

Reply:
Yes, please remove this stub, it is not needed. The version "200" is the minimal version when an Xobj/Xlib appears, e.g. it should be working. Probably there is a different reason for it not picking up "FILE" as a stub. Bud it is clearly a standard FileIO Xobj.

Reply:
Done.